### ISSUE TITLE
mysql 8.0.19 and above supports aliases for ON DUPLICATE KEY UPDATE, VALUES() in replaced row warns

### ISSUE THREAD

**Issue by btilly** [sqlalchemy/sqlalchemy#8626]
### Describe the bug

Starting with mysql 8.0.20 we get a deprecated syntax warning from `on_duplicate_key_update`, and at some point this will break.

The cause and how to fix are explained in https://github.com/sqlalchemy/sqlalchemy/issues/5173#issuecomment-593163758.  However I ran into this bug in the wild, and spent time tracking this down, so I'm putting it into the bug tracker with context for the next person who runs into this error.

Separate note.  I tried to do the `-Werror` flag and that somehow got in the way of connecting to MySQL.  That's not your bug, but is why I didn't give a stack backtrace as requested.

### To Reproduce

```python
from sqlalchemy.dialect.mysql import insert 
from sqlalchemy import func
insert_stmt=insert(t).values(col1=1,col2=2)
upsert_stmt=insert_stmt.on_duplicate_key_update(col2=func.coalesce(insert_stmt.inserted.col2,t.c.col2))
```

### Error

```
/opt/homebrew/lib/python3.9/site-packages/aiomysql/cursors.py:239: Warning: 'VALUES function' is deprecated and will be removed in a future release. Please use an alias (INSERT INTO ... VALUES (...) AS alias) and replace VALUES(col) in the ON DUPLICATE KEY UPDATE clause with alias.col instead
``` 

### Versions

- OS:
- Python:
- SQLAlchemy:
- Database:
- DBAPI (eg: psycopg, cx_oracle, mysqlclient):

**Comment by zzzeek** [comment]
background on MySQL's change:

https://dev.mysql.com/doc/refman/8.0/en/insert-on-duplicate.html

> Beginning with MySQL 8.0.19, it is possible to use an alias for the row, with, optionally, one or more of its columns to be inserted, following the VALUES or SET clause, and preceded by the AS keyword. Using the row alias new, the statement shown previously using VALUES() to access the new column values can be written in the form shown here: 

this will have to be based on version detection and also needs to work for MariaDB.

**Comment by casparwylie** [comment]
Attempting a fix to actually use aliases instead of the deprecated version here
https://github.com/sqlalchemy/sqlalchemy/pull/9210

**Comment by zzzeek** [comment]
> Attempting a fix to actually use aliases instead of the deprecated version here #9210

@casparwylie OK, IIUC this syntax only works in MySQL 8, so assuming that, this compilation logic would need to be conditional for only mysql > 8, not mariadb and not mysql < 8 , is that correct?

**Comment by sqla-tester** [comment]
**Caspar Wylie** has proposed a fix for this issue in the **main** branch:

**Fixes: #8626 compilation of deprecated VALUES() clause instead of newer alias usage** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4412

**Comment by sqla-tester** [comment]
**Caspar Wylie** has proposed a fix for this issue in the **main** branch:

**use mysql 8 syntax for ON DUPLICATE KEY UPDATE** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4412
